### DISCUSSION TITLE
Change data delimiter per field

### DISCUSSION THREAD

**Discussion by nilsenpaul** [engram-design/FeedMe#900] [Ideas]
### Description
We've run into a situation in which we have an (unchangeable) XML-feed that, within a node uses comma's to separate data. Because we want to turn that data into individual tags, we need to set the `dataDelimiter` config setting to ", ". As a result, other field's (in our case a Redactor field's) content first becomes an array and then gets saved as json.

#### Possible solutions:

- Add the ability to change config options per field, in addition to the `feedOptions` config setting.
- Don't try to turn Redactor fields into arrays. 
- Add the ability to 'pre-parse' an input feed with your own PHP callable, from the config file. I would then be able to turn all comma separators into something a bit more unique, before importing.

What are your thoughts on this? 

### Steps to reproduce
Use this input:

```
<JobDescription><![CDATA[<p>Content with, a comma</p>]]></JobDescription>
<Hours>
	<name><![CDATA[Required number of hours]]></name>
	<value><![CDATA[24 - 32 uur, 32 - 36 uur, 36 - 40 uur]]></value>
</Hours
```

Set the `dataDelimiter` to ", " and try to import `JobDescription` to a Redactor field.

### Additional info

- Plugin version: 4.1.1
- Craft version: 3.1.33

**Comment by jamesmacwhite** [comment]
@nilsenpaul I know it's been a while since posting this but one workaround option could be to use the beforeFetchFeed event, to manipulate the XML data before Feed Me processes it, if you can't have the feed data itself changed by whatever provider.

https://docs.craftcms.com/feed-me/v4/developers/events.html#data-fetching-related-events

You could modify the XML node(s) that currently have an ambiguous delimiter and replace with something like `;` or something exotic like `||` so it's less likely to trigger on fields.

I ended up debugging something similar and found the `dataLimiter` setting currently isn't per feedOptions aware either, so that's problematic too for potentially running across other feeds when it doesn't need to.
